agent/07_agent_add_node.sh

```diff
@@ -33,14 +31,36 @@ if [ -f $OCP_DIR/add-node/node.iso ]; then
   rm -f $OCP_DIR/add-node/node.iso
 fi
 
-oc adm node-image create --dir "$OCP_DIR/add-node/" --registry-config "${PULL_SECRET_FILE}" --loglevel=2
+tmpaddnodedir=$(mktemp --tmpdir -d "addnode--XXXXXXXXXX")
+_tmpfiles="$_tmpfiles $tmpaddnodedir"
+cp -rf $OCP_DIR/add-node/* $tmpaddnodedir
+
+case "${AGENT_E2E_TEST_BOOT_MODE}" in
+  "ISO" )
+    oc adm node-image create --dir $tmpaddnodedir --registry-config "${PULL_SECRET_FILE}" --loglevel=2
+
+    for (( n=0; n<${NUM_EXTRA_WORKERS}; n++ ))
+    do
+        sudo virt-xml "${CLUSTER_NAME}_extraworker_${n}" --add-device --disk "$tmpaddnodedir/node.x86_64.iso,device=cdrom,target.dev=sdc"
+        sudo virt-xml "${CLUSTER_NAME}_extraworker_${n}" --edit target=sda --disk="boot_order=1"
+        sudo virt-xml "${CLUSTER_NAME}_extraworker_${n}" --edit target=sdc --disk="boot_order=2" --start
+    done
+    ;;
+
+  "PXE" )
+    oc adm node-image create --pxe --dir $tmpaddnodedir --registry-config "${PULL_SECRET_FILE}" --loglevel=2
+    # Copy the generated PXE artifacts in the tftp server location
+    # The local http server should be running and was started by
+    # day 1 installtion.
+    cp $tmpaddnodedir/boot-artifacts/* ${PXE_SERVER_DIR}
+
+    agent_pxe_boot extraworker $NUM_EXTRA_WORKERS
+    ;;
 
-for (( n=0; n<${NUM_EXTRA_WORKERS}; n++ ))
-do
-    sudo virt-xml "${CLUSTER_NAME}_extraworker_${n}" --add-device --disk "$OCP_DIR/add-node/node.x86_64.iso,device=cdrom,target.dev=sdc"
-    sudo virt-xml "${CLUSTER_NAME}_extraworker_${n}" --edit target=sda --disk="boot_order=1"
-    sudo virt-xml "${CLUSTER_NAME}_extraworker_${n}" --edit target=sdc --disk="boot_order=2" --start
-done
+  "DISKIMAGE" )
```

Comment:
This is not really required?

Reply:
Yes it is not a current requirement. Removing for now.